### PR TITLE
Pkg 1906

### PR DESCRIPTION
**Pull request by markan**
jupyter_server 2.5.0
https://anaconda.atlassian.net/browse/PKG-1906

This is a testing requirement for jupyterlab 3.4.3

* Remove github workflows
* skip s390x
* skip unless python 3.8 or later
* clean up build requirements
* Walked through dependencies and verified against pyproject for version

There is a circular dependency between this and the test dependencies of jupyter_server_ydoc and jupyter_server_terminals

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,57 +1,70 @@
...
 requirements:
   host:
-    - python
-    # To avoid ImportError: cannot import name 'StaticModule' from 'setuptools.config'
-    # we should use jupyter-packaging >=0.12.0
-    - jupyter-packaging >=0.12.0,<1
     - pip
+    - nodejs 18.15.0
+    - hatch-jupyter-builder >=0.8.1
+    - hatchling >=1.11
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: I would suggest using strict pinnings for these build tools

**Review by skupr-anaconda — APPROVED** [review]
LGTM but build tools with strict pinnings

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,57 +1,70 @@
...
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 4ee4f311bd944bcf8060a8b746059571c40f6b8ada1d1e6e51239d26ab23b15c
+  sha256: 9fde612791f716fd34d610cd939704a9639643744751ba66e7ee8fdc9cead07e
 
 build:
   number: 0
-  skip: True  # [py<37]
-  script: {{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv
+  skip: true  # [py<38 or s390x]
```
Comment (akabanovs) [inline recipe/meta.yaml:20]: What is missing that causes us to skip building for s390x?

**Comment by danpetry** [comment]
@akabanovs do you have any remaining changes required to be addressed, or are you able to give this an approval if not?

**Review by akabanovs — APPROVED** [review]
LGTM, but I think the recipe needs # [not win] next to patch in build section, to build on win64.

**Review by skupr-anaconda — APPROVED** [review]

**Review by akabanovs — APPROVED** [review]

**Pull request merged**